hub/core/dataset/__init__.py

```diff
@@ -21,5 +19,8 @@ def dataset_factory(path, *args, **kwargs):
         clz = Dataset
 
     if clz in {Dataset, HubCloudDataset}:
-        return clz(path=path, *args, **kwargs)
+        ds = clz(path=path, *args, **kwargs)
+        if ds.info.get("virtual-datasource", False):
+            ds = ds._get_view()
```

Comment:
We do not expose VDS (dataset with indices) to the user, instead a view to the source dataset is returned.